docs/source/array-creation.rst

```diff
@@ -352,6 +352,45 @@ Files may be retrieved by running `da.from_tiledb` with the same URI, and any
 necessary arguments.
 
 
+Intermediate storage
+--------------------
+
+.. autosummary::
+   store
+
+In some cases, one may wish to store an intermediate result in long term
+storage. This differs from ``persist``, which is mainly used to manage
+intermediate results within Dask that don't necessarily have longevity.
+Also it differs from storing final results as these mark the end of the Dask
+graph. So lack anyway to be reused in other graphs without reloading the data.
+Intermediate storage is mainly useful in cases where the data is needed
+outside of Dask (e.g. on disk, in a database, in the cloud, etc.). It can
+be useful as a checkpoint for long running or flaky computations.
+
+The intermediate storage use case differs from the typical storage use case as
+a Dask Array is returned to the user that represents the result of that
+storage operation. This is typically done by setting the ``store`` function's
+``return_stored`` flag to ``True``. The user can then decide whether the
```

Comment:
```suggestion
``return_stored`` flag to ``True``. 

.. code-block:: python

   x.store()  # stores data, returns nothing
   x = x.store(return_stored=True)  # stores data, returns new dask array backed by that data

The user can then decide whether the
```

I recommend breaking this text with a brief and informative code snippet that gives a sense for what we're after.  Then we can proceed with more details below.